test/conftest.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python3
+# coding=utf-8
+"""
+Variables used across test modules, including the setup and teardown fixture
+that is run before and after every test
+"""
+import os
+import shutil
+import pytest
+
+# sys.path.append("..")
+from ad_logger import ad_logger
+from toolbox import toolbox
+from config import ad_config
+
+# Variables used across test classes
+
+# TODO prevent logging writing to syslog when in testing mode
+
```

Comment:
If you still have a need to disable writing to syslog I suggest something like:

```python
import os
import pytest

@pytest.fixture(scope="session", autouse=True)
def configure_logging_for_tests():
    os.environ['TEST_MODE'] = 'true'
    # Here, configure logging to avoid syslog as shown above
    yield
    # Teardown (if necessary)
    del os.environ['TEST_MODE']
```


Then use something like `if os.getenv('TEST_MODE', 'false').lower() == 'true':` in your code logging code to switch it off.

Reply:
Added to Github issues